### PR TITLE
Add 'duplicatesAndUnique' method to analyze haiku text characters #295

### PR DESCRIPTION
**Pull request by NiveditaMishraCoding**
Add 'duplicatesAndUnique' method to analyze haiku text characters

### CODE REVIEW

**Review by Sweta716 — COMMENTED** [review]
Consider using more descriptive variable names. While chars is clear, charCounts, duplicates, and unique are more explicit and make the code easier to understand.

**Comment by NiveditaMishraCoding** [comment]
> Consider using more descriptive variable names. While chars is clear, charCounts, duplicates, and unique are more explicit and make the code easier to understand.

The variable names were already given. I don’t wanted to make changes on existing variables.

**Review by Sweta716 — APPROVED** [review]

**Review by rachanavishwanath — COMMENTED** [review]
Please rebase and squash the commits

File: company-kata-solutions/src/main/java/org/eclipse/collections/companykata/Order.java [hunk 1]
```diff
@@ -12,13 +12,10 @@
 
 import org.eclipse.collections.api.bag.Bag;
 import org.eclipse.collections.api.bag.sorted.MutableSortedBag;
-import org.eclipse.collections.api.block.function.primitive.DoubleFunction;
 import org.eclipse.collections.impl.block.factory.Comparators;
-import org.eclipse.collections.impl.block.function.AddFunction;
-import org.eclipse.collections.impl.factory.Lists;
 import org.eclipse.collections.impl.factory.SortedBags;
 
-import java.util.List;
```
Comment (rachanavishwanath) [inline company-kata-solutions/src/main/java/org/eclipse/collections/companykata/Order.java:21]: Why are we removing this import? Is it unused?

File: haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java [hunk 2]
```diff
@@ -68,20 +68,20 @@ public record CharCountsDuplicatesUnique(
...
-        // TODO: Find all the unique Characters in the Map (value < 2)
-        // Hint: Look at entrySet, stream, filter, map, Map.Entry.getKey, collect, Collectors.toSet
-        Set<Character> unique = null;
+        Map<Character, Long> chars = this.getHaikuAsChars()
+                .filter(Character::isLetter)               // Filter alphabetic characters
+                .mapToObj(ch -> (char) ch)                // Convert to characters
+                .map(Character::toLowerCase)               // Convert to lowercase
+                .collect(Collectors.groupingBy(c -> c, Collectors.counting()));
+
+        Map<Character, Long> duplicates = chars.entrySet().stream()
```
Comment (rachanavishwanath) [inline haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java:77]: Can you please rename this to duplicateChars

File: haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java [hunk 3]
```diff
@@ -68,20 +68,20 @@ public record CharCountsDuplicatesUnique(
...
+                .filter(entry -> 1L < entry.getValue())    // Filter duplicates
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        Set<Character> unique = chars.entrySet().stream()
```
Comment (rachanavishwanath) [inline haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java:81]: Same as above, please rename to uniqueChars to be specific and readable code

File: haiku-kata-solutions/src/test/java/org/eclipse/collections/haikukata/TextProcessorJDKTest.java [hunk 4]
```diff
@@ -41,8 +41,8 @@ public void distinctLetters()
         Assertions.assertEquals("breakingthoupvmwcdflsy", distinctLetters);
     }
 
-//    @Test // Uncomment once duplicatesAndUnique is implemented for JDK
-//    @Tag("SOLUTION")
+    @Test // Uncomment once duplicatesAndUnique is implemented for JDK
```
Comment (rachanavishwanath) [inline haiku-kata-solutions/src/test/java/org/eclipse/collections/haikukata/TextProcessorJDKTest.java:44]: Please remove the comment as we would not need it post the PR merge

**Review by rzrobin213 — APPROVED** [review]

**Comment by prathasirisha** [comment]
Thank you @NiveditaMishraCoding for your contributions!!

**Pull request merged**